q2_types/per_sample_sequences/_transformer.py

```diff
@@ -94,6 +95,17 @@ def _11(dirfmt: CasavaOneEightLanelessPerSampleDirFmt) \
         dirfmt, SingleLanePerSamplePairedEndFastqDirFmt, parse_lane=False)
 
 
+@plugin.register_transformer
+def _12(dirfmt: SingleLanePerSampleSingleEndFastqDirFmt) \
+        -> CasavaOneEightSingleLanePerSampleDirFmt:
+    casava = CasavaOneEightSingleLanePerSampleDirFmt(mode='w')
+    for file in os.listdir(str(dirfmt)):
```

Comment:
Drive-by comment: check out this example of iterating over a pathlib: https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir